# Genome

Here I just use the blue tit genome, version 1, which Sonja was using - as it already has an MspI cut-sites track bed file made and the bismark genome preparation step is already complete. 

file: `Cyanistes.caeruleus_genome_BT333.1_Illumina.HiSeq_CeleraAssemblerv7.IDBAUDvMAY2014_v1.fa`

Genome Directory: `/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/Genomes/Cyanistes.caeruleus.v1` 

# Pre-Processing

## Single End Libraries

Trim, map, extract Saarland single end libraries. Files already have accessions 

```bash
#!/bin/bash

#SBATCH --get-user-env
#SBATCH --mail-user=merondun@bio.lmu.de
#SBATCH --clusters=biohpc_gen
#SBATCH --partition=biohpc_gen_production
#SBATCH --cpus-per-task=6
#SBATCH --time=200:00:00

genomedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/Genomes/Cyanistes.caeruleus.v1
rawdataRRBS=/dss/dsslegfs01/pr53da/pr53da-dss-0024/rawdata/Bisulfite-Seq/RRBS

basedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing
trimdir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing/scratch
workdir=${basedir}/scratch

#positional arguments
RUN=$1

mkdir ${workdir}

cd ${workdir}

#Trim adapters
trim_galore -j 6 --rrbs --quality 20 --output_dir ${trimdir} --basename ${RUN} $rawdataRRBS/${RUN}*__SE__*fastq.gz

#Map reads
bismark --parallel 6 --output_dir ${workdir} --genome ${genomedir} ${trimdir}/${RUN}_trimmed.fq.gz
mv ${workdir}/${RUN}_trimmed_bismark_bt2.bam ${workdir}/${RUN}_val_1_bismark_bt2_pe.bam
#Extract methylation
bismark_methylation_extractor --parallel 6 --gzip --bedgraph --buffer_size 60G --cytosine_report --genome_folder ${genomedir} --output ${workdir} ${workdir}/${RUN}_val_1_bismark_bt2_pe.bam
```

Submit in parallel with file prefixes:

```bash
cat Saarland_Third.list
B4H0183_BL_ADL_M__SRR24994222__SE__88e82749de07095c29b29f7233500bc6
B4E9350_BL_ADL_F__SRR24994231__SE__c888b50db063baf61fdb01f10d71eb5a
B4P8912_BL_ADL_F__SRR24994225__SE__f377aeb9e018c838ea1229140891be53
B2V8841_BL_ADL_F__SRR24994248__SE__53f6c5893fc6d19bfb78a64e1876a7bd
B4E9350_BL_ADL_F__SRR24994233__SE__b192de37cb9ed817d5c035713943f61b
B4H0271_BL_ADL_M__SRR24994240__SE__a21fafb83083d5637333662f9761b8f9
B2X4825_BL_ADL_M__SRR24994235__SE__609f070f13dca6aed7d8b4a3decfe908
B4H0271_BL_ADL_M__SRR24994238__SE__24c97bf9cb0236dbdf2fa89d7a5b5527
B4H0271_BL_ADL_M__SRR24994239__SE__c1b4acc2824bda2da3d2615ad2fe1230
B4H0051_BL_ADL_F__SRR24994242__SE__0d58370e4b648038eaa0b7401786edb9
B2V8841_BL_ADL_F__SRR24994249__SE__c5dac29a84f849ddf5bfe666b7218acb
B4E9350_BL_ADL_F__SRR24994232__SE__9160461149973f9a610d0dda4eb44d74
B4H0271_BL_ADL_M__SRR24994237__SE__ecb213ba043aefe1546e9435249684d0
B2X4825_BL_ADL_M__SRR24994236__SE__3bf7fe9a90eadfc9cf8a9280f4eb6e72
B4H0051_BL_ADL_F__SRR24994245__SE__3b25ed034efc26493ccd15722e0e53c3
B2X7401_BL_ADL_M__SRR24994250__SE__db6b628df099428d92f3606fbfc2907b
B2X4825_BL_ADL_M__SRR24994234__SE__ced3fa10923961c87e3593aa7257f44b
B4P8912_BL_ADL_F__SRR24994226__SE__8350067f3adf1391e49e438903136836
B4P8912_BL_ADL_F__SRR24994228__SE__33fe1617173ca1186d17b338d17cb107
B4H0051_BL_ADL_F__SRR24994243__SE__bf42079d618ca35e68981fdecadfb280
B4H0051_BL_ADL_F__SRR24994244__SE__84bdda8c366de9e8579776287af4a1a3
B2X7401_BL_ADL_M__SRR24994227__SE__cb6d91f77a4e6030a1691aa4a36d1743
B2V8841_BL_ADL_F__SRR24994221__SE__e0a4e68ce8c7dbf943413f720af7a63b
B2X7401_BL_ADL_M__SRR24994241__SE__d633d8d7611edf6e56841a362e000526
B2X4825_BL_ADL_M__SRR24994247__SE__aadd0bd6c6101da965b7256b2fc40cdb
B4H0183_BL_ADL_M__SRR24994224__SE__1b1424ecf9363d97c2734b258bae708f
B2X7401_BL_ADL_M__SRR24994251__SE__fbdf981d9ec1ffc77d87ba4387b240ab
B4H0183_BL_ADL_M__SRR24994246__SE__36a0a57e961726168e722032eb04e6c1
B4P8912_BL_ADL_F__SRR24994229__SE__6b89adb2a0d150a0a024573c55e586b9
B2V8841_BL_ADL_F__SRR24994220__SE__5e99751b874ecf385421c74781c00feb
B4H0183_BL_ADL_M__SRR24994223__SE__b5c8d503eb1e8563a046e8405f815305
B4E9350_BL_ADL_F__SRR24994230__SE__6358a0f669d43d5a669ae1f2eefe099b


for i in $(cat Saarland_Third.list); do sbatch -J SE_${i} Bismark_SE.sh ${i}; done
```

## Paired End Libraries

Paired end EMseq libraries. These files are NOT currently accessioned, so I will submit jobs with $FILE_R1 $FILE_R2 $PREFIX so that we have sensible names in the end. I simply created the sbatch submission in excel so that we know which files are associated with each prefix:

```bash
#!/bin/bash

#SBATCH --get-user-env
#SBATCH --mail-user=merondun@bio.lmu.de
#SBATCH --clusters=biohpc_gen
#SBATCH --partition=biohpc_gen_production
#SBATCH --cpus-per-task=12
#SBATCH --time=200:00:00

genomedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/Genomes/Cyanistes.caeruleus.v1
rawdataRRBS=/dss/dsslegfs01/pr53da/pr53da-dss-0024/INBOX/KielEmseqjune23/kielfqfiles

basedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing
trimdir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing/scratch
workdir=${basedir}/scratch

#positional arguments
F1=$1
F2=$2
RUN=$3

mkdir ${workdir}
cd ${workdir}

#Trim adapters
trim_galore -j 8 --rrbs --quality 20 --output_dir ${trimdir} --basename ${RUN} --paired $rawdataRRBS/${F1} $rawdataRRBS/${F2}

#Map reads
bismark --parallel 8 --output_dir ${workdir} --genome ${genomedir} -1 ${trimdir}/${RUN}_val_1.fq.gz -2 ${trimdir}/${RUN}_val_2.fq.gz

#Extract methylation
bismark_methylation_extractor --parallel 8 --gzip --bedgraph --buffer_size 60G --cytosine_report --genome_folder ${genomedir} --output ${workdir} ${workdir}/${RUN}_val_1_bismark_bt2_pe.bam

```

And for WGS EMseq:

```bash
#!/bin/bash

#SBATCH --get-user-env
#SBATCH --mail-user=merondun@bio.lmu.de
#SBATCH --clusters=biohpc_gen
#SBATCH --partition=biohpc_gen_production
#SBATCH --cpus-per-task=12
#SBATCH --time=200:00:00

genomedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/Genomes/Cyanistes.caeruleus.v1
rawdataRRBS=/dss/dsslegfs01/pr53da/pr53da-dss-0024/INBOX/KielEmseqjune23/kielfqfiles

basedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing
trimdir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing/scratch
workdir=${basedir}/scratch

#positional arguments
F1=$1
F2=$2
RUN=$3

mkdir ${workdir}
cd ${workdir}

#Trim adapters
trim_galore -j 8 --rrbs --quality 20 --output_dir ${trimdir} --basename ${RUN} --paired $rawdataRRBS/${F1} $rawdataRRBS/${F2}

#Map reads
bismark --parallel 8 --output_dir ${workdir} --genome ${genomedir} -1 ${trimdir}/${RUN}_val_1.fq.gz -2 ${trimdir}/${RUN}_val_2.fq.gz

#Deduplicate
deduplicate_bismark --paired ${workdir}/${RUN}_val_1_bismark_bt2_pe.bam --output_dir ${workdir}

#Extract methylation
bismark_methylation_extractor --parallel 8 --gzip --bedgraph --buffer_size 60G --cytosine_report --genome_folder ${genomedir} --output ${workdir} ${workdir}/${RUN}_val_1_bismark_bt2_pe.deduplicated.bam

```

And submit all:

```bash
#RRBS
sbatch -J R_B2X3614_BL_ADL_F EMSEQ.sh 230200001212-A01_23Feb1212-A01-L2_S2_L004_R1_001.fastq.gz 230200001212-A01_23Feb1212-A01-L2_S2_L004_R2_001.fastq.gz B2X3614_BL_ADL_F_RA
sbatch -J R_B2X3614_BL_ADL_F EMSEQ.sh 230200001212-A02_23Feb1212-A02-L2_S4_L004_R1_001.fastq.gz 230200001212-A02_23Feb1212-A02-L2_S4_L004_R2_001.fastq.gz B2X3614_BL_ADL_F_RB
sbatch -J R_B2X7408_BL_ADL_F EMSEQ.sh 230200001212-B01_23Feb1212-B01-L2_S6_L004_R1_001.fastq.gz 230200001212-B01_23Feb1212-B01-L2_S6_L004_R2_001.fastq.gz B2X7408_BL_ADL_F_RA
sbatch -J R_B2X7408_BL_ADL_F EMSEQ.sh 230200001212-B02_23Feb1212-B02-L2_S8_L004_R1_001.fastq.gz 230200001212-B02_23Feb1212-B02-L2_S8_L004_R2_001.fastq.gz B2X7408_BL_ADL_F_RB
sbatch -J R_B2X7410_BL_ADL_M EMSEQ.sh 230200001212-C01_23Feb1212-C01-L2_S10_L004_R1_001.fastq.gz 230200001212-C01_23Feb1212-C01-L2_S10_L004_R2_001.fastq.gz B2X7410_BL_ADL_M_RA
sbatch -J R_B2X7410_BL_ADL_M EMSEQ.sh 230200001212-C02_23Feb1212-C02-L2_S12_L004_R1_001.fastq.gz 230200001212-C02_23Feb1212-C02-L2_S12_L004_R2_001.fastq.gz B2X7410_BL_ADL_M_RB
sbatch -J R_B2X3498_BL_ADL_M EMSEQ.sh 230200001212-D01_23Feb1212-D01-L2_S14_L004_R1_001.fastq.gz 230200001212-D01_23Feb1212-D01-L2_S14_L004_R2_001.fastq.gz B2X3498_BL_ADL_M_RA
sbatch -J R_B2X3498_BL_ADL_M EMSEQ.sh 230200001212-D02_23Feb1212-D02-L2_S16_L004_R1_001.fastq.gz 230200001212-D02_23Feb1212-D02-L2_S16_L004_R2_001.fastq.gz B2X3498_BL_ADL_M_RB
sbatch -J R_B2X7401_BL_ADL_M EMSEQ.sh 230200001212-E01_23Feb1212-E01-L2_S18_L004_R1_001.fastq.gz 230200001212-E01_23Feb1212-E01-L2_S18_L004_R2_001.fastq.gz B2X7401_BL_ADL_M_RA
sbatch -J R_B4E9350_BL_ADL_F EMSEQ.sh 230200001212-F01_23Feb1212-F01-L2_S20_L004_R1_001.fastq.gz 230200001212-F01_23Feb1212-F01-L2_S20_L004_R2_001.fastq.gz B4E9350_BL_ADL_F_RA
sbatch -J R_B2V8841_BL_ADL_F EMSEQ.sh 230200001212-G01_23Feb1212-G01-L2_S22_L004_R1_001.fastq.gz 230200001212-G01_23Feb1212-G01-L2_S22_L004_R2_001.fastq.gz B2V8841_BL_ADL_F_RA
sbatch -J R_B2X4825_BL_ADL_M EMSEQ.sh 230200001212-H01_23Feb1212-H01-L2_S24_L004_R1_001.fastq.gz 230200001212-H01_23Feb1212-H01-L2_S24_L004_R2_001.fastq.gz B2X4825_BL_ADL_M_RA
#WGBS
sbatch -J W_B2X3614_BL_ADL_F EMSEQ_WGS.sh 230200001212-A01_23Feb1212-A01-L1_S1_L004_R1_001.fastq.gz 230200001212-A01_23Feb1212-A01-L1_S1_L004_R2_001.fastq.gz B2X3614_BL_ADL_F_WA
sbatch -J W_B2X3614_BL_ADL_F EMSEQ_WGS.sh 230200001212-A02_23Feb1212-A02-L1_S3_L004_R1_001.fastq.gz 230200001212-A02_23Feb1212-A02-L1_S3_L004_R2_001.fastq.gz B2X3614_BL_ADL_F_WB
sbatch -J W_B2X7408_BL_ADL_F EMSEQ_WGS.sh 230200001212-B01_23Feb1212-B01-L1_S5_L004_R1_001.fastq.gz 230200001212-B01_23Feb1212-B01-L1_S5_L004_R2_001.fastq.gz B2X7408_BL_ADL_F_WA
sbatch -J W_B2X7408_BL_ADL_F EMSEQ_WGS.sh 230200001212-B02_23Feb1212-B02-L1_S7_L004_R1_001.fastq.gz 230200001212-B02_23Feb1212-B02-L1_S7_L004_R2_001.fastq.gz B2X7408_BL_ADL_F_WB
sbatch -J W_B2X7410_BL_ADL_M EMSEQ_WGS.sh 230200001212-C01_23Feb1212-C01-L1_S9_L004_R1_001.fastq.gz 230200001212-C01_23Feb1212-C01-L1_S9_L004_R2_001.fastq.gz B2X7410_BL_ADL_M_WA
sbatch -J W_B2X7410_BL_ADL_M EMSEQ_WGS.sh 230200001212-C02_23Feb1212-C02-L1_S11_L004_R1_001.fastq.gz 230200001212-C02_23Feb1212-C02-L1_S11_L004_R2_001.fastq.gz B2X7410_BL_ADL_M_WB
sbatch -J W_B2X3498_BL_ADL_M EMSEQ_WGS.sh 230200001212-D01_23Feb1212-D01-L1_S13_L004_R1_001.fastq.gz 230200001212-D01_23Feb1212-D01-L1_S13_L004_R2_001.fastq.gz B2X3498_BL_ADL_M_WA
sbatch -J W_B2X3498_BL_ADL_M EMSEQ_WGS.sh 230200001212-D02_23Feb1212-D02-L1_S15_L004_R1_001.fastq.gz 230200001212-D02_23Feb1212-D02-L1_S15_L004_R2_001.fastq.gz B2X3498_BL_ADL_M_WB
sbatch -J W_B2X7401_BL_ADL_M EMSEQ_WGS.sh 230200001212-E01_23Feb1212-E01-L1_S17_L004_R1_001.fastq.gz 230200001212-E01_23Feb1212-E01-L1_S17_L004_R2_001.fastq.gz B2X7401_BL_ADL_M_WA
sbatch -J W_B4E9350_BL_ADL_F EMSEQ_WGS.sh 230200001212-F01_23Feb1212-F01-L1_S19_L004_R1_001.fastq.gz 230200001212-F01_23Feb1212-F01-L1_S19_L004_R2_001.fastq.gz B4E9350_BL_ADL_F_WA
sbatch -J W_B2V8841_BL_ADL_F EMSEQ_WGS.sh 230200001212-G01_23Feb1212-G01-L1_S21_L004_R1_001.fastq.gz 230200001212-G01_23Feb1212-G01-L1_S21_L004_R2_001.fastq.gz B2V8841_BL_ADL_F_WA
sbatch -J W_B2X4825_BL_ADL_M EMSEQ_WGS.sh 230200001212-H01_23Feb1212-H01-L1_S23_L004_R1_001.fastq.gz 230200001212-H01_23Feb1212-H01-L1_S23_L004_R2_001.fastq.gz B2X4825_BL_ADL_M_WA

```

## Existing PE-WGBS

Also map and extract the original WGBS libraries from Kiel. 

```bash
#!/bin/bash

#SBATCH --get-user-env
#SBATCH --mail-user=merondun@bio.lmu.de
#SBATCH --clusters=biohpc_gen
#SBATCH --partition=biohpc_gen_production
#SBATCH --cpus-per-task=12
#SBATCH --time=200:00:00

genomedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/Genomes/Cyanistes.caeruleus.v1
rawdataWGBS=/dss/dsslegfs01/pr53da/pr53da-dss-0024/rawdata/Bisulfite-Seq/WGBS

basedir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing
trimdir=/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023FEB/processing/scratch
workdir=${basedir}/scratch

#positional arguments
RUN=$1

mkdir ${workdir}
cd ${workdir}

F1=$( realpath ${rawdataWGBS}/${RUN}*__R1__*gz)
F2=$( realpath ${rawdataWGBS}/${RUN}*__R2__*gz)

#Trim adapters
trim_galore -j 8 --rrbs --quality 20 --output_dir ${trimdir} --basename ${RUN} --paired ${F1} ${F2}

#Map reads
bismark --parallel 9 --output_dir ${workdir} --genome ${genomedir} -1 ${trimdir}/${RUN}_val_1.fq.gz -2 ${trimdir}/${RUN}_val_2.fq.gz

#Deduplicate
deduplicate_bismark --paired ${workdir}/${RUN}_val_1_bismark_bt2_pe.bam --output_dir ${workdir}

#Extract methylation
bismark_methylation_extractor --parallel 9 --gzip --bedgraph --buffer_size 60G --cytosine_report --genome_folder ${genomedir} --output ${workdir} ${workdir}/${RUN}_val_1_bismark_bt2_pe.deduplicated.bam

```

And the files:

```bash
cat WGBS.list
B2X4825_BL_ADL_M__SRR10606819
B2X4825_BL_ADL_M__SRR10606820
B2X4930_BL_ADL_F__SRR10606817
B2X4930_BL_ADL_F__SRR10606818
B2X6537_BL_CHK_M__SRR10606815
B2X6537_BL_CHK_M__SRR10606816
B2X6540_BL_CHK_M__SRR10606813
B2X6540_BL_CHK_M__SRR10606814
B2X7441_BL_CHK_M__SRR10606810
B2X7441_BL_CHK_M__SRR10606811
B2X7449_BL_CHK_M__SRR10606808
B2X7449_BL_CHK_M__SRR10606809
C2R0242_BL_ADL_F__SRR10606806
C2R0242_BL_ADL_F__SRR10606807
C2R0553_BL_ADL_M__SRR10606804
C2R0553_BL_ADL_M__SRR10606805

for i in $(cat WGBS.list); do sbatch -J WGBS_${i} WGBS_KIEL.sh ${i}; done
```

# Basic Summaries

I got raw and filtered 5mC CpGs just by counting the .cov.gz files directly `3.Parse_Coverage_Counts.sh`: 

```bash
#!/bin/bash

#SBATCH --get-user-env
#SBATCH --mail-user=merondun@bio.lmu.de
#SBATCH --clusters=biohpc_gen
#SBATCH --partition=biohpc_gen_normal
#SBATCH --cpus-per-task=2
#SBATCH --time=48:00:00

rm Coverage_Statistics.txt

for i in $(ls *cov.gz | sed 's/.cov.gz//g'); do

abv0=$(zcat ${i}.cov.gz | wc -l)
abv10=$(zcat ${i}.cov.gz | awk '$5+$6>=10' | wc -l)
echo -e "${i}\t${abv0}\t${abv10}" >> Coverage_Statistics.txt

done
```

## Alignment & Global 5mC

Alignment rates and global 5mC comes directly from alignment reports: `2.Parse_Bismark_Alignement_Reports.sh` 

```
# Takes bismark alignment_report.txt as input. The $ID will be the string preceding '.alignment_report.txt', so rename your files accordingly
echo -e "ID\ttotal_sequences\tunique_hits\tmapping_rate\ttotal_cytosines\tcpg_rate\tchg_rate\tchh_rate" > processed_reports.txt

for file in *.alignment_report.txt
do
    ID=$(basename "$file" .alignment_report.txt)
    total_sequences=$(grep "analysed in total:" "$file" | awk '{print $NF}')
    unique_hits=$(grep "unique best hit" "$file" | awk '{print $NF}') #
    mapping_rate=$(grep "Mapping efficiency:" "$file" | awk '{print $3}')
    total_cytosines=$(grep "Total number of C's analysed:" "$file" | awk '{print $6}')
    cpg_rate=$(grep "C methylated in CpG context:" "$file" | awk '{print $6}')
    chg_rate=$(grep "C methylated in CHG context:" "$file" | awk '{print $6}')
    chh_rate=$(grep "C methylated in CHH context:" "$file" | awk '{print $6}')

    # Append to the output file
    echo -e "$ID\t$total_sequences\t$unique_hits\t$mapping_rate\t$total_cytosines\t$cpg_rate\t$chg_rate\t$chh_rate" >> processed_reports.txt
done

```

## M-Bias 

This script just takes a directory full of M-Bias.txt files, the output of bismark cpg extractor, which have this terrible format:

```bash
head B2X7642_BL_ADL_M__SRR22581286.M-Bias.txt
CpG context (R1)
================
position        count methylated        count unmethylated      % methylation   coverage
1       1083772 896287  54.73   1980059
2       5120    2921    63.67   8041
3       17637   14622   54.67   32259

```

And output like this `1.Parse_Bismark_M-Bias_Files.sh` 

```bash
for file in *M-Bias.txt
do
    outfile="${file%.*}_processed.txt"
    context=""
    readtype=""
    prefix=$(echo ${file} | sed 's/\..*//g' )
    rm -f "$outfile"
    while IFS= read -r line
    do
        if [[ $line == *"context"* ]]; then
            context=$(echo $line | awk '{print $1}')
            if [[ $line == *"R1"* ]]; then
                readtype="R1"
            elif [[ $line == *"R2"* ]]; then
                readtype="R2"
            else
                readtype="SE"
            fi
        elif [[ $line == [1-9]* ]]; then
            echo -e "$line\t$context\t$readtype\t$prefix" >> "$outfile"
        fi
    done < "$file"
done
```

And will output many files like this, which can be analyzed in R:

```bash
head B3Y6742_BL_ADL_F__SRR20216544.M-Bias_processed.txt
1       1552027 1486984 51.07   3039011 CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
2       6752    6424    51.24   13176   CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
3       21721   32150   40.32   53871   CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
4       13723   25082   35.36   38805   CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
5       29056   21148   57.88   50204   CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
6       51709   40444   56.11   92153   CpG     R1      B3Y6742_BL_ADL_F__SRR20216544
```

## Plot

Very basic plot just from metadata

```R
setwd('/dss/dsshome1/lxc07/di39dux/merondun/cyanistes_methylation')
.libPaths('~/mambaforge/envs/r/lib/R/library')
library(tidyverse)
library(viridis)
library(meRo) #install_github("merondun/meRo")

d = read.table('metadata_n467_pilots.txt',header=TRUE,sep='\t',comment.char='') %>% as_tibble
d = d %>% mutate_at(c('Number10XCpGs','NumberRawCpGs','MappingRate'),as.numeric) %>% drop_na(Number10XCpGs) 
dp = d %>% 
  pivot_longer(c(MappingRate,NumberRawCpGs,Number10XCpGs)) %>%
  group_by(Batch,name) %>% 
  sum_stats(value) %>% 
  ggplot(aes(x=Batch,y=mean,ymax=mean+sd,ymin=mean-sd,col=Batch))+
  geom_point()+
  ylab('Mean +/- SD')+
  scale_color_viridis(discrete=TRUE)+
  facet_grid(name~.,scales='free')+
  geom_errorbar()+
  theme_bw()

png('Summary_Statistics.png',height=5,width=7,units='in',res=600)
dp
dev.off()

```

# Overlap cov.gz Files

Takes the bismark cov.gz files, outputs a total CpG file, a CpG file with max-missing 20% individuals by site, and a discrete-binned coverage information file which can be used to plot for trends. Run on a high memory node (n=10 works), will take about 5-6 hours to run with n=500 samples including WGBS. 

```R
library(tidyverse)
library(matrixStats)

# Path with cov.gz files
setwd('/dss/dsslegfs01/pr53da/pr53da-dss-0024/projects/2022_EpigeneticsTits/2023JULY/bismark_cov')

# Get a list of all .cov.gz files in the working directory
files <- list.files(path=".", pattern="*.cov.gz", full.names=TRUE, recursive=FALSE)

# Initialize empty lists to store names and coverage data
allnames <- NULL
cov <- NULL

# Loop through all .cov.gz files
for (samp in files) {
  cat('Reading in file: ',samp,'\n') # Display current file being read

  # Read in .cov.gz file
  ztab <- gzfile(samp,'rt');
  tab <- read.table(ztab,header=F);
  names(tab) = c('chr','start','end','Percent_5mC','countM','countU')

  # Extract name for new variable, trim all the garbage
  name <- gsub('./','',samp)
  name <- gsub('_val_1.*','',name); name <- gsub('_trimmed.*','',name); name <- gsub('.cov.gz','',name)

  # Calculate quick coverage and %5mc summary statistics per sample and store them
  tab = tab %>% mutate(Coverage=countM+countU,ID=name)
  c1 <- tab %>% mutate(Discrete = cut(Coverage, breaks = c(0,10,25,100,Inf), right = F, labels = c('Below10x','Below25x','Below100x','Above100x'))) %>% group_by(Discrete,ID) %>% dplyr::count()
  c2 <- tab %>% mutate(Discrete = cut(Percent_5mC, breaks = c(-1,10,90,101), right = F, labels = c('Below10p','Below90p','Above90p'))) %>% group_by(Discrete,ID) %>% dplyr::count()
  c1$Variable <- 'Coverage'
  c2$Variable <- 'Percent_5mC'
  cc <- rbind(c1,c2)
  cov <- rbind(cov,cc)

  # Keep only positions with 10x coverage or higher ### IMPORTANT!
  tab <- subset(tab, Coverage > 9)
  tab$site <- paste0(tab$chr,'_',tab$start)
  tab$Percent_5mC <- tab$Percent_5mC/100

  # Grab only site / countM / countU / %5mC
  tab2 <- tab %>% select(site,Percent_5mC)
  names(tab2) <- c('site',name);

  cat('Saved to variable: ',name,'\n') # Indicate current file has been processed

  # Assign data to a new variable
  assign(paste0(name),tab2)

  # Add current sample to total list
  allnames <- unique(c(allnames,name))
}

# Get names of all the cov.gz files loaded
vars <- mget(allnames)

# Merge all the CpGs across all the samples, keeping all sites even if covered by only 1 sample
master <- Reduce(function(x,y) merge(x = x, y = y, by=c('site'),all=TRUE),vars)

# Filter according to missingness. If there are more than 'missing_samples' NAs at a site, drop the site
missingness_threshold = 0.2
colcount = ncol(master) - 1
filtered = master %>% mutate(
  NAs = rowSums(is.na(.)),
  NAp = NAs / colcount) %>%
  as_tibble %>%
  filter(NAp <= 0.2) %>%
  select(-c(NAs,NAp))

write.table(master,file='../overlapped_cpgs/5mC_Data_Pilot_10x.txt',quote=F,sep='\t',row.names=FALSE)
write.table(cov,file='../overlapped_cpgs/Coverage-Sample_Statistics_10x.txt',quote=F,sep='\t',row.names=F)

#write file with missingness filters
write.table(filtered,file='../overlapped_cpgs/5mC_Data_Pilot_10x-MM2.txt',quote=F,sep='\t',row.names=FALSE)
```


